synapse/lib/cell.py

```diff
@@ -2360,8 +2422,16 @@ async def initFromArgv(cls, argv, outp=None):
         conf.setConfFromOpts(opts)
         conf.setConfFromEnvs()
 
+        # resolve cell.yaml configs so we have a complete config
+        yamlpath = s_common.genpath(opts.dirn, 'cell.yaml')
+        if os.path.isfile(yamlpath):
+            for k, v in s_common.yamlload(yamlpath).items():
+                conf.setdefault(k, v)
```

Comment:
We should call `conf.reqConfValid()` before procAhaSvc. In the event of a bad config, we'd be stopping prior to using the `aha:provision` value.